proxy/src/markdown-extensions/image-internal.ts

```diff
@@ -0,0 +1,37 @@
+import type { TokenizerAndRendererExtension } from 'marked'
+
+export type ImageInternalToken = {
+  type: 'image-internal'
+  raw: string
+  href: string
+  text: string
+  linkUrl: string | null
+}
+
+export const imageInternalExtension: TokenizerAndRendererExtension = {
+  name: 'image-internal',
+  level: 'inline',
+
+  tokenizer(src, _tokens): ImageInternalToken | undefined {
+    const rule = /^@@@\s*image-internal\s*\n!\[([^\]]+)\]\(([^)]+)\)\s*\n@@@/
+    const match = rule.exec(src)
+
+    if (match) {
+      const [raw, text, href, linkUrl] = match
+
+      return {
+        type: 'image-internal',
+        raw: raw,
+        text: text.trim(),
+        href: href,
```

Comment:
Consider calling this `src`?

Reply:
This is just currently matching the other image tokeniser, and I think it is to avoid confusion between the src parameter in the tokeniser.